src/components/App/App.js

```diff
@@ -0,0 +1,596 @@
+// -------------------------------
+// COMPONENT IMPORTS
+// -------------------------------
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import AboutUs from "../AboutUs/AboutUs";
+import TermsAndConditions from "../TermsAndConditions/TermsAndConditions";
+import Footer from "../Footer/Footer";
+import Profile from "../Profile/Profile";
+
+// -------------------------------
+// MODAL COMPONENT IMPORTS
+// -------------------------------
+import LogInModal from "../LogInModal/LogInModal";
+import WelcomeModal from "../WelcomeModal/WelcomeModal";
+import RegisterModal from "../RegisterModal/RegisterModal";
+import EditProfileModal from "../EditProfileModal/EditProfileModal";
+import TinyPopup from "../TinyPopUp/TinyPopUp";
+
+// -------------------------------
+// CONTEXT AND UTILITY IMPORTS
+// -------------------------------
+import ProtectedRoute from "../ProtectedRoute/ProtectedRoute";
+import CurrentUserContext from "../../contexts/CurrentUserContext";
+import * as ai from "../../utils/OracleApi";
+import * as auth from "../../utils/Auth";
+import * as api from "../../utils/Api";
+
+function App() {
+  // -------------------------------
+  // STATE MANAGEMENT
+  // -------------------------------
+
+  // MODAL AND LOADING STATES
+  const [activeModal, setActiveModal] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  // USER AUTHENTICATION STATES
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [currentUser, setCurrentUser] = useState({});
+  const [inputError, setInputError] = useState("");
+  const [tinyPopup, setTinyPopup] = useState({ isVisible: false, message: "" });
+
+  // ORACLE FUNCTIONALITY STATES
+  const [oracleResponse, setOracleResponse] = useState("");
+  const [isOracleProcessingSTT, setIsOracleProcessingSTT] = useState(false);
+  const [isOracleProcessingTTS, setIsOracleProcessingTTS] = useState(false);
+  const [isOraclePlayingAudio, setIsOraclePlayingAudio] = useState(false);
+  const [oracleReadings, setOracleReadings] = useState([]);
+
+  // MEDIA RECORDING STATES
+  const [mediaRecorder, setMediaRecorder] = useState(null);
+  const [isRecording, setIsRecording] = useState(false);
+  const [isReadingCompleted, setIsReadingCompleted] = useState(false);
+  const [isUserTalking, setIsUserTalking] = useState(false);
+
+  // MICROPHONE ACTIVATION STATES
+  const [isMicActivated, setIsMicActivated] = useState(false);
+  const [isMicActivationPopupVisible, setIsMicActivationPopupVisible] =
+    useState(false);
+
+  const history = useHistory();
+
+  // -------------------------
+  // MODALS
+  // -------------------------
+
+  useEffect(() => {
+    if (!activeModal) return;
+    const handleEscClose = (e) => {
+      if (e.key === "Escape") {
+        handleCloseModal();
+      }
+    };
+    document.addEventListener("keydown", handleEscClose);
+    return () => {
+      document.removeEventListener("keydown", handleEscClose);
+    };
+  }, [activeModal]);
```

Comment:
The `Escape` handler is not needed in `App` because you have it inside component `Modal`